staging/src/k8s.io/apiserver/pkg/server/options/etcd.go

```diff
@@ -375,6 +375,47 @@ func (s *EtcdOptions) addEtcdHealthEndpoint(c *server.Config) error {
 		return readyCheck()
 	}))
 
+	if len(s.EtcdServersOverrides) != 0 {
+		if err := s.addOverrideEtcdHealthEndpoint(c); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (s *EtcdOptions) addOverrideEtcdHealthEndpoint(c *server.Config) error {
+	var overrideSet = sets.NewString()
+
+	for _, override := range s.EtcdServersOverrides {
+		tokens := strings.Split(override, "#")
+		// multi overrides may point to the same servers
+		// example: ["apps/deployments#s1.example.com;s2.example.com","apps/replicasets#s1.example.com;s2.example.com"]
+		overrideSet.Insert(tokens[1])
+	}
+
+	for i, override := range overrideSet.List() {
```

Comment:
Converting set to list will lose the ordering which will result in a poor user experience since the override doesn't show the resource name and it only shows an index.
I'd suggest preserve the ordering and then dedup. Something like:
```
for _, override := range s.EtcdServersOverrides {
    tokens := strings.Split(override, "#")
    if overrideSet.Contains(tokens[1]) {
        continue
    }
    overrideSet.Insert(tokens[1])

    // Then add health check and readiness check for 
    // fmt.Sprintf("etcd-override-%d", len(overrideSet))
}
```

Reply:
Miss that. Updated.